discovery/syncer_test.go

```diff
@@ -2411,3 +2512,205 @@ func TestGossipSyncerMaxChannelRangeReplies(t *testing.T) {
+			require.GreaterOrEqual(
+				t, attemptCount, 1,
+				"state handler was not called - test "+
+					"setup issue",
+			)
+			require.LessOrEqual(
+				t, attemptCount, 5,
+				"too many attempts (%d) - endless loop "+
+					"not fixed",
+				attemptCount,
+			)
+
+			// Verify the syncer exits cleanly without hanging.
+			assertSyncerExitsCleanly(t, syncer, 2*time.Second)
+		})
+	}
+}
+
+// TestGossipSyncerProcessChanRangeReplyError tests that errors from
+// processChanRangeReply cause the goroutine to exit cleanly. This test uses
+// a different approach than the state handler tests because the
+// waitingQueryRangeReply state waits in a select rather than looping.
+func TestGossipSyncerProcessChanRangeReplyError(t *testing.T) {
+	t.Parallel()
+
+	hID := lnwire.NewShortChanIDFromInt(10)
+	msgChan := make(chan []lnwire.Message, 20)
+
+	// Create custom channel series.
+	chanSeries := newMockChannelGraphTimeSeries(hID)
+
+	cfg := gossipSyncerCfg{
+		channelSeries:          chanSeries,
+		encodingType:           defaultEncoding,
+		chunkSize:              defaultChunkSize,
+		batchSize:              defaultChunkSize,
+		noSyncChannels:         false,
+		noReplyQueries:         true,
+		noTimestampQueryOption: false,
+		sendMsg: func(_ context.Context, _ bool,
+			msgs ...lnwire.Message) error {
+
+			msgChan <- msgs
+			return nil
+		},
+		bestHeight: func() uint32 {
+			return latestKnownHeight
+		},
+		markGraphSynced:          func() {},
+		maxQueryChanRangeReplies: maxQueryChanRangeReplies,
+		timestampQueueSize:       10,
+	}
+
+	syncerSema := make(chan struct{}, 1)
+	syncerSema <- struct{}{}
+
+	syncer := newGossipSyncer(cfg, syncerSema)
+
+	// Start the syncer which will send a QueryChannelRange message.
+	syncer.Start()
+	defer syncer.Stop()
+
+	// Wait for the syncer to send its QueryChannelRange message.
+	var query *lnwire.QueryChannelRange
+	select {
+	case msgs := <-msgChan:
+		require.Len(t, msgs, 1)
+
+		var ok bool
+		query, ok = msgs[0].(*lnwire.QueryChannelRange)
+		require.True(t, ok, "expected QueryChannelRange message")
+
+	case <-time.After(time.Second):
+		t.Fatal("expected query message")
+	}
+
+	// Send 10 malformed replies. With the fix, only the first one should be
+	// processed before the goroutine exits. Without the fix, all would be
+	// processed (goroutine stays in waitingQueryRangeReply state).
+	for i := 0; i < 10; i++ {
+		malformedReply := &lnwire.ReplyChannelRange{
+			ChainHash: query.ChainHash,
+			// FirstBlockHeight is before query range (invalid).
+			FirstBlockHeight: query.FirstBlockHeight - 100,
+			NumBlocks:        query.NumBlocks,
+			Complete:         1,
+			ShortChanIDs:     []lnwire.ShortChannelID{},
+		}
+
+		// Send the malformed reply.
+		err := syncer.ProcessQueryMsg(malformedReply, nil)
+		require.NoError(t, err)
+
+		// Brief delay to let goroutine process.
+		time.Sleep(20 * time.Millisecond)
+	}
+
+	// With the fix: goroutine exits after first error, so remaining 9
```

Comment:
same here the comment should not have "fix in it" its super relative, couple of commit down the road we don't know to what this is referring to